src/Lecture2.hs

```diff
@@ -190,8 +416,15 @@ verify that.
 >>> merge [1, 2, 4] [3, 7]
 [1,2,3,4,7]
 -}
-merge :: [Int] -> [Int] -> [Int]
-merge = error "TODO"
+merge :: Ord a => [a] -> [a] -> [a]
+-- is this cheating? :) 
+-- merge xs ys = sort (xs ++ ys)
+merge xs ys = case (xs, ys) of
+    ([], _) -> ys
+    (_, []) -> xs
+    (x : xs', y : ys') ->
+        if x < y then x : merge xs' ys
+        else y : merge xs ys'
```

Comment:
Absolutely correct solution! 💎 
You can make it more performant by checking if both heads of the given lists are equal. In that case, you're able to reduce the size of each list by 1 on each recursive call instead of reducing the size of only a single list ✂️